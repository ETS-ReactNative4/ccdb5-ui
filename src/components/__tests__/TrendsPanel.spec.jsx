import configureMockStore from 'redux-mock-store'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import ReduxTrendsPanel, {
  TrendsPanel,
  mapDispatchToProps
} from '../Trends/TrendsPanel'
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import thunk from 'redux-thunk'

jest.mock( 'britecharts', () => {
  const props = [
    'brush', 'line', 'tooltip', 'margin', 'backgroundColor', 'colorSchema',
    'enableLabels', 'labelsSize', 'labelsTotalCount', 'labelsNumberFormat',
    'outerPadding', 'percentageAxisToMaxRatio', 'yAxisLineWrapLimit',
    'dateRange', 'yAxisPaddingBetweenChart', 'width', 'wrapLabels', 'height',
    'on', 'initializeVerticalMarker', 'isAnimated', 'tooltipThreshold', 'grid',
    'aspectRatio', 'dateLabel', 'shouldShowDateInTitle', 'topicLabel', 'title'
  ]

  const mock = {}

  for ( let i = 0; i < props.length; i++ ) {
    const propName = props[i]
    mock[propName] = jest.fn().mockImplementation( () => {
      return mock
    } )
  }

  return mock
} )

jest.mock( 'd3', () => {
  const props = [
    'select', 'each', 'node', 'getBoundingClientRect', 'width', 'datum', 'call',
    'remove', 'selectAll'
  ]

  const mock = {}

  for ( let i = 0; i < props.length; i++ ) {
    const propName = props[i]
    mock[propName] = jest.fn().mockImplementation( () => {
      return mock
    } )
  }

  // set narrow width value for 100% test coverage
  mock.width = 100

  return mock
} )

function setupEnzyme( { focus, overview, lens, subLens } ) {
  const props = {
    focus,
    lenses: [ 'Foo', 'Baz', 'Bar' ],
    intervals: [ 'Month', 'Quarter', 'Nickel', 'Day' ],
    overview,
    lens,
    subLens,
    onChartType: jest.fn(),
    onInterval: jest.fn(),
    onLens: jest.fn()
  }
  const target = shallow( <TrendsPanel { ...props }/> )
  return target
}

function setupSnapshot( { chartType,
                          company,
                          focus,
                          lens,
                          subLens,
                          trendsDateWarningEnabled,
                          width }) {
  const middlewares = [ thunk ]
  const mockStore = configureMockStore( middlewares )
  const store = mockStore( {
    aggs: {
      doc_count: 10000,
      total: 1000
    },
    query: {
      company,
      date_received_min: "2018-01-01T00:00:00.000Z",
      date_received_max: "2020-01-01T00:00:00.000Z",
      lens,
      subLens,
      trendsDateWarningEnabled
    },
    trends: {
      chartType,
      colorMap: { "Complaints": "#ADDC91", "Other": "#a2a3a4" },
      focus,
      lens,
      results: {
        "dateRangeBrush": [
          {
            "date": new Date( "2020-01-01T00:00:00.000Z" ),
            "value": 26413
          }, {
            "date": new Date( "2020-02-01T00:00:00.000Z" ),
            "value": 25096
          }, {
            "date": new Date( "2020-03-01T00:00:00.000Z" ),
            "value": 29506
          }, {
            "date": new Date( "2020-04-01T00:00:00.000Z" ),
            "value": 35112
          }, {
            "date": new Date( "2020-05-01T00:00:00.000Z" ),
            "value": 9821
          } ],
        "dateRangeLine": {
          "dataByTopic": [ {
            "topic": "Complaints",
            "topicName": "Complaints",
            "dashed": false,
            "show": true,
            "dates": [
              { "date": "2020-03-01T00:00:00.000Z", "value": 29506 },
              { "date": "2020-04-01T00:00:00.000Z", "value": 35112 },
              { "date": "2020-05-01T00:00:00.000Z", "value": 9821 }
            ]
          } ]
        },
        issue: [ { name: 'adg', value: 123 } ],
        product: [ { name: 'adg', value: 123 } ],
        company: [ { name: 'adg', value: 123 } ]
      },
      total: 10000
    },
    view: {
      width
    }
  } )

  return renderer.create(
    <Provider store={ store }>
      <IntlProvider locale="en">
        <ReduxTrendsPanel/>
      </IntlProvider>
    </Provider>
  )
}

describe( 'component:TrendsPanel', () => {
  describe( 'Snapshots', () => {
    let params
    beforeEach(()=>{
      params = {
        chartType: 'line',
        company: false,
        focus: '',
        lens: 'Overview',
        printMode: false,
        showMobileFilters: false,
        subLens: 'sub_product',
        trendsDateWarningEnabled: false,
        width: 1000
      }
    })

    it( 'renders company Overlay without crashing', () => {
      params.company = []
      params.lens = 'Company'

      const target = setupSnapshot( params )
      const tree = target.toJSON()
      expect( tree ).toMatchSnapshot()
    } )

    it( 'renders lineChart Overview without crashing', () => {
      params.lens = 'Overview'
      const target = setupSnapshot( params )
      const tree = target.toJSON()
      expect( tree ).toMatchSnapshot()
    } )

    it( 'renders area without crashing', () => {
      params.chartType = 'area'
      params.lens = 'Product'
      params.subLens = 'sub_product'
      const target = setupSnapshot( params )
      const tree = target.toJSON()
      expect( tree ).toMatchSnapshot()
    } )

    it( 'renders date warning without crashing', () => {
      params.trendsDateWarningEnabled = true
      const target = setupSnapshot( params )
      const tree = target.toJSON()
      expect( tree ).toMatchSnapshot()
    } )

    it( 'renders Focus without crashing', () => {
      params.focus = 'Yippe'
      params.lens = 'Product'
      const target = setupSnapshot( params )
      const tree = target.toJSON()
      expect( tree ).toMatchSnapshot()
    } )

    it( 'renders print mode without crashing', () => {
      params.printMode = true
      const target = setupSnapshot( params )
      const tree = target.toJSON()
      expect( tree ).toMatchSnapshot()
    } )

    it( 'renders mobile filters without crashing', () => {
      params.width = 600
      const target = setupSnapshot( params )
      const tree = target.toJSON()
      expect( tree ).toMatchSnapshot()
    } )

    it( 'renders external Tooltip without crashing', () => {
      params.lens = 'Product'
      const target = setupSnapshot( params )
      const tree = target.toJSON()
      expect( tree ).toMatchSnapshot()
    } )
  } )

  describe( 'helpers', () => {
    describe( 'areaChartTitle', () => {
      let params
      beforeEach( () => {
        params = {
          focus: '',
          overview: false,
          lens: 'Overview',
          subLens: 'sub_product'
        }
      } )
      it( 'gets area chart title - Overview', () => {
        params.overview = true
        const target = setupEnzyme( params )
        expect( target.instance()._areaChartTitle() )
          .toEqual( 'Complaints by date CFPB received' )
      } )

      it( 'gets area chart title - Data Lens', () => {
        params.lens = 'Something'
        const target = setupEnzyme( params )
        expect( target.instance()._areaChartTitle() )
          .toEqual( 'Complaints by something by date CFPB received' )
      } )

      it( 'gets area chart title - Focus', () => {
        params.focus = 'Hello'
        params.lens = 'Product'
        const target = setupEnzyme( params )
        expect( target.instance()._areaChartTitle() )
          .toEqual( 'Complaints by sub-products by date CFPB received' )
      } )
    } )
  } )

  describe( 'mapDispatchToProps', () => {
    it( 'hooks into changeChartType', () => {
      const dispatch = jest.fn()
      mapDispatchToProps( dispatch )
        .onChartType( {
          target: {
            value: 'foo'
          }
        } )
      expect( dispatch.mock.calls.length ).toEqual( 1 )
    } )
    it( 'hooks into changeDateInterval', () => {
      const dispatch = jest.fn()
      mapDispatchToProps( dispatch )
        .onInterval( {
          target: {
            value: 'foo'
          }
        } )
      expect( dispatch.mock.calls.length ).toEqual( 1 )
    } )
    it( 'hooks into changeDataLens', () => {
      const dispatch = jest.fn()
      mapDispatchToProps( dispatch )
        .onLens( {
          target: {
            value: 'foo'
          }
        } )
      expect( dispatch.mock.calls.length ).toEqual( 1 )
    } )

    it( 'hooks into dismissWarning', () => {
      const dispatch = jest.fn()
      mapDispatchToProps( dispatch ).onDismissWarning()
      expect( dispatch.mock.calls ).toEqual( [
        [ {
          requery: 'REQUERY_NEVER',
          type: 'TRENDS_DATE_WARNING_DISMISSED'
        } ]
      ] )
    } )
  } )
} )
