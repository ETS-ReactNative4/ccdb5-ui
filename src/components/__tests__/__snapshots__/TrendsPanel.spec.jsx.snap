// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:TrendsPanel renders area without crashing 1`] = `
<section
  className="trends-panel"
>
  <div>
    <summary
      className="action-bar"
      id="search-summary"
    >
      <div>
        <h2>
          Showing 
          <span>
            1,000
          </span>
           matches out of 
          <span>
            10,000
          </span>
           total complaints
        </h2>
      </div>
      <div
        className="layout-row"
      >
        <h3
          className="h4 flex-all export-results"
        >
          <button
            className="a-btn a-btn__link"
            data-gtm_ignore="true"
            onClick={[Function]}
          >
            Export data
          </button>
          <button
            className="a-btn a-btn__link print-preview"
            data-gtm_ignore="true"
            onClick={[Function]}
          >
            <svg
              className="cf-icon-svg"
              viewBox="0 0 761.9 1200"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M203.7 889.9h355v50h-355zM203.7 788.4h355v50h-355z"
              />
              <path
                d="M670.9 368.4L477.3 174.8H122.1c-16.5 0-30 13.5-30 30V500H70c-38.5 0-70 31.5-70 70v305.5c0 38.5 31.5 70 70 70h22.1v60.5c0 16.5 13.5 30 30 30h518.8c16.5 0 30-13.5 30-30v-60.5h21c38.5 0 70-31.5 70-70V569.9c0-38.5-31.5-70-70-70h-21V368.4zm-60 607.6H152.1V742.7h458.8V976zM425 234.8v146.4c0 21.1 17.3 38.4 38.4 38.4h147.5V500H152.1V234.8H425zm237.9 403.9c-16.6 0-30-13.4-30-30s13.4-30 30-30 30 13.4 30 30-13.4 30-30 30z"
              />
            </svg>
            Print
          </button>
        </h3>
      </div>
    </summary>
  </div>
  <div
    className="layout-row refine-bar"
  >
    <section
      className="filter-panel-toggle"
    >
      <div
        className="m-btn-group"
      >
        <p>
           
        </p>
        <button
          className="a-btn"
          onClick={[Function]}
          title="Filter results"
        >
          Filter results
        </button>
      </div>
    </section>
    <section
      className="cf-select"
    >
      <label
        className="u-visually-hidden"
        htmlFor="choose-lens"
      >
        Aggregate complaints by
      </label>
      <p>
        Aggregate by
      </p>
      <select
        id="choose-lens"
        onChange={[MockFunction]}
        value="Overview"
      >
        <option
          value="Overview"
        >
          Overview
        </option>
        <option
          value="Company"
        >
          Company
        </option>
        <option
          value="Product"
        >
          Product
        </option>
        <option
          value="Issue"
        >
          Issue
        </option>
      </select>
    </section>
    <span
      className="separator"
    />
    <section
      className="cf-select"
    >
      <label
        className="u-visually-hidden"
        htmlFor="choose-interval"
      >
        Choose the Date Interval
      </label>
      <p>
        Date Interval
      </p>
      <select
        id="choose-interval"
        onChange={[MockFunction]}
      >
        <option
          value="Day"
        >
          Day
        </option>
        <option
          value="Week"
        >
          Week
        </option>
        <option
          value="Month"
        >
          Month
        </option>
        <option
          value="Quarter"
        >
          Quarter
        </option>
        <option
          value="Year"
        >
          Year
        </option>
      </select>
    </section>
    <section
      className="date-ranges m-btn-group"
    >
      <p>
        Date range (Click to modify range)
      </p>
      <button
        className="a-btn date-selector range-3m"
        onClick={[Function]}
      >
        3m
      </button>
      <button
        className="a-btn date-selector range-6m"
        onClick={[Function]}
      >
        6m
      </button>
      <button
        className="a-btn date-selector range-1y"
        onClick={[Function]}
      >
        1y
      </button>
      <button
        className="a-btn date-selector range-3y"
        onClick={[Function]}
      >
        3y
      </button>
      <button
        className="a-btn date-selector range-All"
        onClick={[Function]}
      >
        All
      </button>
    </section>
  </div>
  <div
    className="layout-row"
  >
    <section
      className="chart"
    >
      <div>
        <h2>
          Complaints by date received
        </h2>
        <div
          id="stacked-area-chart"
        />
      </div>
      <div
        id="brush-chart"
      />
    </section>
  </div>
  <div
    className="row-chart-section"
  >
    <h3>
      Product by highest complaint volume
    </h3>
    <div
      id="row-chart-product"
    />
  </div>
  <div
    className="row-chart-section"
  >
    <h3>
      Issue by highest complaint volume
    </h3>
    <div
      id="row-chart-issue"
    />
  </div>
</section>
`;

exports[`component:TrendsPanel renders external Tooltip without crashing 1`] = `
<section
  className="trends-panel external-tooltip"
>
  <div>
    <summary
      className="action-bar"
      id="search-summary"
    >
      <div>
        <h2>
          Showing 
          <span>
            1,000
          </span>
           matches out of 
          <span>
            10,000
          </span>
           total complaints
        </h2>
      </div>
      <div
        className="layout-row"
      >
        <h3
          className="h4 flex-all export-results"
        >
          <button
            className="a-btn a-btn__link"
            data-gtm_ignore="true"
            onClick={[Function]}
          >
            Export data
          </button>
          <button
            className="a-btn a-btn__link print-preview"
            data-gtm_ignore="true"
            onClick={[Function]}
          >
            <svg
              className="cf-icon-svg"
              viewBox="0 0 761.9 1200"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M203.7 889.9h355v50h-355zM203.7 788.4h355v50h-355z"
              />
              <path
                d="M670.9 368.4L477.3 174.8H122.1c-16.5 0-30 13.5-30 30V500H70c-38.5 0-70 31.5-70 70v305.5c0 38.5 31.5 70 70 70h22.1v60.5c0 16.5 13.5 30 30 30h518.8c16.5 0 30-13.5 30-30v-60.5h21c38.5 0 70-31.5 70-70V569.9c0-38.5-31.5-70-70-70h-21V368.4zm-60 607.6H152.1V742.7h458.8V976zM425 234.8v146.4c0 21.1 17.3 38.4 38.4 38.4h147.5V500H152.1V234.8H425zm237.9 403.9c-16.6 0-30-13.4-30-30s13.4-30 30-30 30 13.4 30 30-13.4 30-30 30z"
              />
            </svg>
            Print
          </button>
        </h3>
      </div>
    </summary>
  </div>
  <div
    className="layout-row refine-bar"
  >
    <section
      className="filter-panel-toggle"
    >
      <div
        className="m-btn-group"
      >
        <p>
           
        </p>
        <button
          className="a-btn"
          onClick={[Function]}
          title="Filter results"
        >
          Filter results
        </button>
      </div>
    </section>
    <section
      className="cf-select"
    >
      <label
        className="u-visually-hidden"
        htmlFor="choose-lens"
      >
        Aggregate complaints by
      </label>
      <p>
        Aggregate by
      </p>
      <select
        id="choose-lens"
        onChange={[MockFunction]}
        value="Product"
      >
        <option
          value="Overview"
        >
          Overview
        </option>
        <option
          value="Company"
        >
          Company
        </option>
        <option
          value="Product"
        >
          Product
        </option>
        <option
          value="Issue"
        >
          Issue
        </option>
      </select>
    </section>
    <span
      className="separator"
    />
    <section
      className="cf-select"
    >
      <label
        className="u-visually-hidden"
        htmlFor="choose-interval"
      >
        Choose the Date Interval
      </label>
      <p>
        Date Interval
      </p>
      <select
        id="choose-interval"
        onChange={[MockFunction]}
      >
        <option
          value="Day"
        >
          Day
        </option>
        <option
          value="Week"
        >
          Week
        </option>
        <option
          value="Month"
        >
          Month
        </option>
        <option
          value="Quarter"
        >
          Quarter
        </option>
        <option
          value="Year"
        >
          Year
        </option>
      </select>
    </section>
    <section
      className="date-ranges m-btn-group"
    >
      <p>
        Date range (Click to modify range)
      </p>
      <button
        className="a-btn date-selector range-3m"
        onClick={[Function]}
      >
        3m
      </button>
      <button
        className="a-btn date-selector range-6m"
        onClick={[Function]}
      >
        6m
      </button>
      <button
        className="a-btn date-selector range-1y"
        onClick={[Function]}
      >
        1y
      </button>
      <button
        className="a-btn date-selector range-3y"
        onClick={[Function]}
      >
        3y
      </button>
      <button
        className="a-btn date-selector range-All"
        onClick={[Function]}
      >
        All
      </button>
    </section>
    <span
      className="separator"
    />
    <section
      className="chart-toggles m-btn-group"
    >
      <p>
        Chart Type
      </p>
      <button
        className="a-btn toggle line"
        onClick={[Function]}
      >
        <svg
          data-name="line chart"
          id="line-chart-icon"
          viewBox="0 0 27 21"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            line chart icon
          </title>
          <g
            className="cls-1"
          >
            <line
              className="cls-2"
              x1="12"
              x2="2"
              y1="6.27"
              y2="19"
            />
            <line
              className="cls-2"
              x1="22.9"
              x2="14.1"
              y1="14"
              y2="6"
            />
            <path
              className="cls-3"
              d="M13,4a1,1,0,1,1-1,1,1,1,0,0,1,1-1m0-1a2,2,0,1,0,2,2,2,2,0,0,0-2-2Z"
            />
            <path
              className="cls-3"
              d="M24,14a1,1,0,1,1-1,1,1,1,0,0,1,1-1m0-1a2,2,0,1,0,2,2,2,2,0,0,0-2-2Z"
            />
          </g>
          <line
            className="cls-2"
            x1="12"
            x2="2"
            y1="16.45"
            y2="11"
          />
          <line
            className="cls-2"
            x1="23"
            x2="14"
            y1="6"
            y2="16"
          />
          <path
            className="cls-3"
            d="M13,16a1,1,0,1,1-1,1,1,1,0,0,1,1-1m0-1a2,2,0,1,0,2,2,2,2,0,0,0-2-2Z"
          />
          <path
            className="cls-3"
            d="M24,4a1,1,0,1,1-1,1,1,1,0,0,1,1-1m0-1a2,2,0,1,0,2,2,2,2,0,0,0-2-2Z"
          />
        </svg>
      </button>
      <button
        className="a-btn toggle area"
        onClick={[Function]}
      >
        <svg
          data-name="area chart"
          id="area-chart-icon"
          viewBox="0 0 27 21"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            area chart icon
          </title>
          <polygon
            className="cls-1"
            points="27 11 23.48 14 15.26 9 7.04 17 0 15 0 21 27 21 27 11"
          />
          <g
            className="cls-2"
          >
            <polygon
              className="cls-1"
              points="6.32 14.79 14.92 6.46 23.36 11.58 27 8.5 27 1 23.48 4 15.26 1 7.04 7 0 9 0 12.94 6.32 14.79"
            />
          </g>
        </svg>
      </button>
    </section>
  </div>
  <div
    className="layout-row"
  >
    <section
      className="chart"
    >
      <div>
        <h2>
          Complaints by date received
        </h2>
        <div
          id="line-chart"
        />
      </div>
      <div
        id="brush-chart"
      />
    </section>
  </div>
  <div
    className="tabbed-navigation lens"
  >
    <h2>
      Product trends for selected criteria
    </h2>
    <section>
      <button
        className="tab sub_product"
        onClick={[Function]}
      >
        Sub-products
      </button>
      <button
        className="tab issue active"
        onClick={[Function]}
      >
        Issues
      </button>
    </section>
  </div>
  <div
    className="row-chart-section"
  >
    <h3>
      Some title SubLens
    </h3>
    <div
      id="row-chart-Product"
    />
  </div>
</section>
`;

exports[`component:TrendsPanel renders line without crashing 1`] = `
<section
  className="trends-panel"
>
  <div>
    <summary
      className="action-bar"
      id="search-summary"
    >
      <div>
        <h2>
          Showing 
          <span>
            1,000
          </span>
           matches out of 
          <span>
            10,000
          </span>
           total complaints
        </h2>
      </div>
      <div
        className="layout-row"
      >
        <h3
          className="h4 flex-all export-results"
        >
          <button
            className="a-btn a-btn__link"
            data-gtm_ignore="true"
            onClick={[Function]}
          >
            Export data
          </button>
          <button
            className="a-btn a-btn__link print-preview"
            data-gtm_ignore="true"
            onClick={[Function]}
          >
            <svg
              className="cf-icon-svg"
              viewBox="0 0 761.9 1200"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M203.7 889.9h355v50h-355zM203.7 788.4h355v50h-355z"
              />
              <path
                d="M670.9 368.4L477.3 174.8H122.1c-16.5 0-30 13.5-30 30V500H70c-38.5 0-70 31.5-70 70v305.5c0 38.5 31.5 70 70 70h22.1v60.5c0 16.5 13.5 30 30 30h518.8c16.5 0 30-13.5 30-30v-60.5h21c38.5 0 70-31.5 70-70V569.9c0-38.5-31.5-70-70-70h-21V368.4zm-60 607.6H152.1V742.7h458.8V976zM425 234.8v146.4c0 21.1 17.3 38.4 38.4 38.4h147.5V500H152.1V234.8H425zm237.9 403.9c-16.6 0-30-13.4-30-30s13.4-30 30-30 30 13.4 30 30-13.4 30-30 30z"
              />
            </svg>
            Print
          </button>
        </h3>
      </div>
    </summary>
  </div>
  <div
    className="layout-row refine-bar"
  >
    <section
      className="filter-panel-toggle"
    >
      <div
        className="m-btn-group"
      >
        <p>
           
        </p>
        <button
          className="a-btn"
          onClick={[Function]}
          title="Filter results"
        >
          Filter results
        </button>
      </div>
    </section>
    <section
      className="cf-select"
    >
      <label
        className="u-visually-hidden"
        htmlFor="choose-lens"
      >
        Aggregate complaints by
      </label>
      <p>
        Aggregate by
      </p>
      <select
        id="choose-lens"
        onChange={[MockFunction]}
        value="Overview"
      >
        <option
          value="Overview"
        >
          Overview
        </option>
        <option
          value="Company"
        >
          Company
        </option>
        <option
          value="Product"
        >
          Product
        </option>
        <option
          value="Issue"
        >
          Issue
        </option>
      </select>
    </section>
    <span
      className="separator"
    />
    <section
      className="cf-select"
    >
      <label
        className="u-visually-hidden"
        htmlFor="choose-interval"
      >
        Choose the Date Interval
      </label>
      <p>
        Date Interval
      </p>
      <select
        id="choose-interval"
        onChange={[MockFunction]}
      >
        <option
          value="Day"
        >
          Day
        </option>
        <option
          value="Week"
        >
          Week
        </option>
        <option
          value="Month"
        >
          Month
        </option>
        <option
          value="Quarter"
        >
          Quarter
        </option>
        <option
          value="Year"
        >
          Year
        </option>
      </select>
    </section>
    <section
      className="date-ranges m-btn-group"
    >
      <p>
        Date range (Click to modify range)
      </p>
      <button
        className="a-btn date-selector range-3m"
        onClick={[Function]}
      >
        3m
      </button>
      <button
        className="a-btn date-selector range-6m"
        onClick={[Function]}
      >
        6m
      </button>
      <button
        className="a-btn date-selector range-1y"
        onClick={[Function]}
      >
        1y
      </button>
      <button
        className="a-btn date-selector range-3y"
        onClick={[Function]}
      >
        3y
      </button>
      <button
        className="a-btn date-selector range-All"
        onClick={[Function]}
      >
        All
      </button>
    </section>
  </div>
  <div
    className="layout-row"
  >
    <section
      className="chart"
    >
      <div>
        <h2>
          Complaints by date received
        </h2>
        <div
          id="line-chart"
        />
      </div>
      <div
        id="brush-chart"
      />
    </section>
  </div>
  <div
    className="row-chart-section"
  >
    <h3>
      Product by highest complaint volume
    </h3>
    <div
      id="row-chart-product"
    />
  </div>
  <div
    className="row-chart-section"
  >
    <h3>
      Issue by highest complaint volume
    </h3>
    <div
      id="row-chart-issue"
    />
  </div>
</section>
`;

exports[`component:TrendsPanel renders mobile filters without crashing 1`] = `
<section
  className="trends-panel"
>
  <div>
    <summary
      className="action-bar"
      id="search-summary"
    >
      <div>
        <h2>
          Showing 
          <span>
            1,000
          </span>
           matches out of 
          <span>
            10,000
          </span>
           total complaints
        </h2>
      </div>
      <div
        className="layout-row"
      >
        <h3
          className="h4 flex-all export-results"
        >
          <button
            className="a-btn a-btn__link"
            data-gtm_ignore="true"
            onClick={[Function]}
          >
            Export data
          </button>
          <button
            className="a-btn a-btn__link print-preview"
            data-gtm_ignore="true"
            onClick={[Function]}
          >
            <svg
              className="cf-icon-svg"
              viewBox="0 0 761.9 1200"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M203.7 889.9h355v50h-355zM203.7 788.4h355v50h-355z"
              />
              <path
                d="M670.9 368.4L477.3 174.8H122.1c-16.5 0-30 13.5-30 30V500H70c-38.5 0-70 31.5-70 70v305.5c0 38.5 31.5 70 70 70h22.1v60.5c0 16.5 13.5 30 30 30h518.8c16.5 0 30-13.5 30-30v-60.5h21c38.5 0 70-31.5 70-70V569.9c0-38.5-31.5-70-70-70h-21V368.4zm-60 607.6H152.1V742.7h458.8V976zM425 234.8v146.4c0 21.1 17.3 38.4 38.4 38.4h147.5V500H152.1V234.8H425zm237.9 403.9c-16.6 0-30-13.4-30-30s13.4-30 30-30 30 13.4 30 30-13.4 30-30 30z"
              />
            </svg>
            Print
          </button>
        </h3>
      </div>
    </summary>
  </div>
  <div />
  <div
    className="layout-row refine-bar"
  >
    <section
      className="filter-panel-toggle"
    >
      <div
        className="m-btn-group"
      >
        <p>
           
        </p>
        <button
          className="a-btn"
          onClick={[Function]}
          title="Filter results"
        >
          Filter results
        </button>
      </div>
    </section>
    <section
      className="cf-select"
    >
      <label
        className="u-visually-hidden"
        htmlFor="choose-lens"
      >
        Aggregate complaints by
      </label>
      <p>
        Aggregate by
      </p>
      <select
        id="choose-lens"
        onChange={[MockFunction]}
        value="Overview"
      >
        <option
          value="Overview"
        >
          Overview
        </option>
        <option
          value="Company"
        >
          Company
        </option>
        <option
          value="Product"
        >
          Product
        </option>
        <option
          value="Issue"
        >
          Issue
        </option>
      </select>
    </section>
    <span
      className="separator"
    />
    <section
      className="cf-select"
    >
      <label
        className="u-visually-hidden"
        htmlFor="choose-interval"
      >
        Choose the Date Interval
      </label>
      <p>
        Date Interval
      </p>
      <select
        id="choose-interval"
        onChange={[MockFunction]}
      >
        <option
          value="Day"
        >
          Day
        </option>
        <option
          value="Week"
        >
          Week
        </option>
        <option
          value="Month"
        >
          Month
        </option>
        <option
          value="Quarter"
        >
          Quarter
        </option>
        <option
          value="Year"
        >
          Year
        </option>
      </select>
    </section>
    <section
      className="date-ranges m-btn-group"
    >
      <p>
        Date range (Click to modify range)
      </p>
      <button
        className="a-btn date-selector range-3m"
        onClick={[Function]}
      >
        3m
      </button>
      <button
        className="a-btn date-selector range-6m"
        onClick={[Function]}
      >
        6m
      </button>
      <button
        className="a-btn date-selector range-1y"
        onClick={[Function]}
      >
        1y
      </button>
      <button
        className="a-btn date-selector range-3y"
        onClick={[Function]}
      >
        3y
      </button>
      <button
        className="a-btn date-selector range-All"
        onClick={[Function]}
      >
        All
      </button>
    </section>
  </div>
  <div
    className="layout-row"
  >
    <section
      className="chart"
    >
      <div>
        <h2>
          Complaints by date received
        </h2>
        <div
          id="line-chart"
        />
      </div>
      <div
        id="brush-chart"
      />
    </section>
  </div>
  <div
    className="row-chart-section"
  >
    <h3>
      Product by highest complaint volume
    </h3>
    <div
      id="row-chart-product"
    />
  </div>
  <div
    className="row-chart-section"
  >
    <h3>
      Issue by highest complaint volume
    </h3>
    <div
      id="row-chart-issue"
    />
  </div>
</section>
`;

exports[`component:TrendsPanel renders print mode without crashing 1`] = `
<section
  className="trends-panel"
>
  <div>
    <summary
      className="action-bar"
      id="search-summary"
    >
      <div>
        <h2>
          Showing 
          <span>
            1,000
          </span>
           matches out of 
          <span>
            10,000
          </span>
           total complaints
        </h2>
      </div>
      <div
        className="layout-row"
      >
        <h3
          className="h4 flex-all export-results"
        >
          <button
            className="a-btn a-btn__link"
            data-gtm_ignore="true"
            onClick={[Function]}
          >
            Export data
          </button>
          <button
            className="a-btn a-btn__link print-preview"
            data-gtm_ignore="true"
            onClick={[Function]}
          >
            <svg
              className="cf-icon-svg"
              viewBox="0 0 761.9 1200"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M203.7 889.9h355v50h-355zM203.7 788.4h355v50h-355z"
              />
              <path
                d="M670.9 368.4L477.3 174.8H122.1c-16.5 0-30 13.5-30 30V500H70c-38.5 0-70 31.5-70 70v305.5c0 38.5 31.5 70 70 70h22.1v60.5c0 16.5 13.5 30 30 30h518.8c16.5 0 30-13.5 30-30v-60.5h21c38.5 0 70-31.5 70-70V569.9c0-38.5-31.5-70-70-70h-21V368.4zm-60 607.6H152.1V742.7h458.8V976zM425 234.8v146.4c0 21.1 17.3 38.4 38.4 38.4h147.5V500H152.1V234.8H425zm237.9 403.9c-16.6 0-30-13.4-30-30s13.4-30 30-30 30 13.4 30 30-13.4 30-30 30z"
              />
            </svg>
            Print
          </button>
        </h3>
      </div>
    </summary>
  </div>
  <div
    className="layout-row refine-bar"
  >
    <section
      className="filter-panel-toggle"
    >
      <div
        className="m-btn-group"
      >
        <p>
           
        </p>
        <button
          className="a-btn"
          onClick={[Function]}
          title="Filter results"
        >
          Filter results
        </button>
      </div>
    </section>
    <section
      className="cf-select"
    >
      <label
        className="u-visually-hidden"
        htmlFor="choose-lens"
      >
        Aggregate complaints by
      </label>
      <p>
        Aggregate by
      </p>
      <select
        id="choose-lens"
        onChange={[MockFunction]}
        value="Overview"
      >
        <option
          value="Overview"
        >
          Overview
        </option>
        <option
          value="Company"
        >
          Company
        </option>
        <option
          value="Product"
        >
          Product
        </option>
        <option
          value="Issue"
        >
          Issue
        </option>
      </select>
    </section>
    <span
      className="separator"
    />
    <section
      className="cf-select"
    >
      <label
        className="u-visually-hidden"
        htmlFor="choose-interval"
      >
        Choose the Date Interval
      </label>
      <p>
        Date Interval
      </p>
      <select
        id="choose-interval"
        onChange={[MockFunction]}
      >
        <option
          value="Day"
        >
          Day
        </option>
        <option
          value="Week"
        >
          Week
        </option>
        <option
          value="Month"
        >
          Month
        </option>
        <option
          value="Quarter"
        >
          Quarter
        </option>
        <option
          value="Year"
        >
          Year
        </option>
      </select>
    </section>
    <section
      className="date-ranges m-btn-group"
    >
      <p>
        Date range (Click to modify range)
      </p>
      <button
        className="a-btn date-selector range-3m"
        onClick={[Function]}
      >
        3m
      </button>
      <button
        className="a-btn date-selector range-6m"
        onClick={[Function]}
      >
        6m
      </button>
      <button
        className="a-btn date-selector range-1y"
        onClick={[Function]}
      >
        1y
      </button>
      <button
        className="a-btn date-selector range-3y"
        onClick={[Function]}
      >
        3y
      </button>
      <button
        className="a-btn date-selector range-All"
        onClick={[Function]}
      >
        All
      </button>
    </section>
  </div>
  <div
    className="layout-row"
  >
    <section
      className="chart"
    >
      <div>
        <h2>
          Complaints by date received
        </h2>
        <div
          id="line-chart"
        />
      </div>
      <div
        id="brush-chart"
      />
    </section>
  </div>
  <div
    className="row-chart-section"
  >
    <h3>
      Product by highest complaint volume
    </h3>
    <div
      id="row-chart-product"
    />
  </div>
  <div
    className="row-chart-section"
  >
    <h3>
      Issue by highest complaint volume
    </h3>
    <div
      id="row-chart-issue"
    />
  </div>
</section>
`;
